graylog2-web-interface/src/components/sidecars/configurations/CopyCollectorModal.jsx

```diff
@@ -44,52 +43,40 @@ class CopyCollectorModal extends React.Component {
   };
 
   _save = () => {
-    const collector = this.state;
+    const { errorMessage, id, name } = this.state;
+    const { copyCollector } = this.props;
 
-    if (!collector.error) {
-      this.props.copyCollector(this.state.id, this.state.name, this._saved);
+    if (!errorMessage) {
+      copyCollector(id, name, this._saved);
     }
   };
 
   _changeName = (event) => {
-    const nextName = event.target.value;
-    this.setState({ name: nextName });
+    const { collector, validateCollector } = this.props;
+    const name = event.target.value;
+    this.setState({ name, errorMessage: undefined });
 
-    const nextCollector = lodash.cloneDeep(this.props.collector);
-    nextCollector.name = nextName;
+    const nextCollector = lodash.cloneDeep(collector);
+    nextCollector.name = name;
     nextCollector.id = '';
 
-    this.props.validateCollector(nextCollector).then((validation) => {
-      let errorMessage = '';
-      if (validation.errors.name) {
-        errorMessage = validation.errors.name[0];
-      }
-      this.setState({ error: validation.failed, error_message: errorMessage });
+    validateCollector(nextCollector).then((validation) => {
+      const [errorMessage] = validation.errors.name;
```

Comment:
The `validation` parameter will not always have those fields, and it may even be an empty object. Trying to destructure it will result in an error.

```
Unhandled rejection TypeError: Invalid attempt to destructure non-iterable instance
at _nonIterableRest (http://localhost:8080/7c9d6ab7-32.js:1273:37)
at _slicedToArray (http://localhost:8080/7c9d6ab7-32.js:1271:99)
at http://localhost:8080/7c9d6ab7-32.js:1392:37
```